x-pack/legacy/plugins/index_management/public/app/components/mappings_editor/lib/utils.test.ts

```diff
@@ -62,4 +62,77 @@ describe('utils', () => {
       expect(isStateValid(components)).toBe(false);
     });
   });
+
+  describe('extractMappingsDefinition', () => {
+    test('should detect that the mappings has multiple types and return null', () => {
+      const mappings = {
+        type1: {
+          properties: {
+            name1: {
+              type: 'keyword',
+            },
+          },
+        },
+        type2: {
+          properties: {
+            name2: {
+              type: 'keyword',
+            },
+          },
+        },
+      };
+
+      expect(extractMappingsDefinition(mappings)).toBe(null);
+    });
+
+    test('should detect that the mappings has one type under the "_doc" type', () => {
+      const mappings = {
```

Comment:
After chatting with Gordon, I realized that in 7.x users may still have index templates and indices with single custom types. So in this example, type1 would also be acceptable in place of _doc. Might be worth adding another test case?

Reply:
The "_doc" name is arbitrary, the logic does not look at the name but at how many "types" are declared. So "_doc" or "type1" is the same. I will update the test title to make that clearer.